src/lib.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::kbc_modules::{KbcCheckInfo, KbcInstance, KbcModuleList};
+use anyhow::*;
+use std::collections::HashMap;
+
+mod kbc_modules;
+
+/// Attestation Agent (AA for short) is a rust library crate for attestation procedure
+/// in confidential containers. It provides kinds of service APIs that need to make
+/// requests to the Relying Party (Key Broker Service) in Confidential Containers,
+/// and establishes an attestation and connection between the corresponding Key Broker
+/// Client (KBC) and KBS, so as to obtain the trusted services or resources of KBS.
+///
+/// # Example
+///
+/// ```rust
+/// use attestation_agent::AttestationAgent;
+/// use attestation_agent::ServiceAPIs;
+///
+/// let mut aa = AttestationAgent::new();
+///
+/// let key_result = aa.decrypt(
+///     "sample_kbc".to_string(),
+///     "https://xxxxx".to_string(),
+///     "example_annotation".to_string()
+/// );
+/// ```
+
+/// `ServiceAPIs` defines the service APIs of attestation agent that need to make requests
+///  to the Relying Party (Key Broker Service) in Confidential Containers.
+///
+/// For every service API, the `kbc_name` and `kbs_uri` is necessary, `kbc_name` tells
+/// attestation agent which KBC module it should use and `kbs_uri` specifics the KBS address.
+pub trait ServiceAPIs {
```

Comment:
`ServiceAPIs` seems a bit vague. Maybe we could call it `AttestationAPI` or `AgentEndpoints` or something.